Connection.php

```diff
@@ -459,8 +459,8 @@ protected function httpRequest($method, $url, $requestBody = null, $raw = false)
             CURLOPT_FORBID_REUSE   => false,
         ];
 
-        if (!empty($this->curlOptions)) {
-            $options = array_merge($options, $this->curlOptions);
+        foreach ($this->curlOptions as $key => $value) {
```

Comment:
How would that produce different results?

Reply:
Arrays with numeric keys are renumbered with `array_merge`. Using foreach the keys are preserved.

Reply:
Constants like `CURLOPT_SSL_VERIFYPEER` have integer values, so array like this is merged incorrectly
`[CURLOPT_SSL_VERIFYPEER => false, CURLOPT_SSL_VERIFYHOST => false]`

Reply:
Alright!

Reply:
A nicer solution would be
```
if (!empty($this->curlOptions)) {
    $options = $options + $this->curlOptions;
}
```

Reply:
@HagnMarkus this would not work if you wanted to override already defined option like `CURLOPT_USERAGENT`, value from `curlOptions` would be discarded as it's already defined in `$options`

however it would work the other way around `$options = $this->curlOptions + $options;`